petl/io/json.py

```diff
@@ -16,7 +16,7 @@
 from petl.util.base import data, Table, dicts as _dicts, iterpeek
 
 
-def fromjson(source, *args, **kwargs):
+def fromjson(source=None, *args, **kwargs):
```

Comment:
>  Code scanning / Prospector (reported by Codacy) 
>
> Keyword argument before variable positional arguments list in the definition of fromjson function (keyword-arg-before-vararg)

This wouldn't break existing code?

Maybe [read_source_from_arg](https://github.com/petl-developers/petl/blob/master/petl/io/sources.py#L465)  can be taught to read even when stdin is passed as argument as in:

``` python
    table1 = etl.fromjson(sys.stdin, header=["foo", "bar"])
    print(table1)
```

Reply:
I might be wrong, but I don't think it should break existing code.

All these assertions pass -

```python
>>> def f(s, *a, **kw):
...     return (s, a, kw)
... 
>>> def fn(s=None, *a, **kw):
...     return (s, a, kw)
... 
>>> assert f("hello") == fn("hello")
>>> assert f(None) == fn(None)
>>> assert f("hello", 1, 2) == fn("hello", 1, 2)
>>> assert f("hello", k=1) == fn("hello", k=1)
>>> assert f(s="hello") == fn(s="hello")
>>> assert f("hello", 1, 2, k=1, j=2) == fn("hello", 1, 2, k=1, j=2)
```

re. `read_source_from_arg()` - my interest is in the `petl` executable, i.e., I would love for syntax like this to work in shell: `echo '{"a":2, "b":4}' | petl 'fromjson(lines=True)'`.

What do you think?

Reply:
I think that:

1. Using petl as a command line tool is an interesting use case and we should give it some love by:
   a. Adding more examples in the [documentation](https://petl.readthedocs.io/en/stable/intro.html#petl-executable).
   b. Mention this usage pattern in the repository Readme/Frontage
2. Currently [fromjson()](https://petl.readthedocs.io/en/stable/io.html#petl.io.json.fromjson) source argument is not consistent  with other functions like:
   a.  [tojson()](https://petl.readthedocs.io/en/stable/io.html#petl.io.json.tojson) and [tojsonarrays()](https://petl.readthedocs.io/en/stable/io.html#petl.io.json.tojsonarrays)
   b. [fromcsv()](https://petl.readthedocs.io/en/stable/io.html#petl.io.csv.fromcsv)
3. We value not breaking existing code.
   a. But I haven't had the time to check if this is the case yet.
   b. If not it will be a good change for API consistency
   c. I would love to hear more opinions about this change.